bin/SymfonyPolyfillCleanup.php

```diff
@@ -0,0 +1,234 @@
+ */
+class SymfonyPolyfillCleanup {
+
+	/**
+	 * Display debug output.
+	 *
+	 * @var boolean
+	 */
+	protected $debug = false;
+
+	/**
+	 * @var Event Composer event.
+	 */
+	protected $event = null;
+
+	/**
+	 * @var string The path of the Symfony library.
+	 */
+	protected $path = null;
+
+	/**
+	 * @var array The list of symfony polyfills.
+	 */
+	protected const LIST_OF_POLYFILLS = [
+		'polyfill-intl-normalizer',
+		'polyfill-mbstring',
+	];
+
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Event|null  $event Composer event.
+	 * @param string|null $path  Path of the Symfony library.
+	 */
+	public function __construct( Event $event = null, string $path = null ) {
+		$this->event = $event;
+		$this->path  = $path ?: dirname( __DIR__ ) . '/vendor/symfony/';
+	}
+
+	/**
+	 * Get list of Sympfony polyfills.
+	 *
+	 * @return array List of sympfony polyfills.
+	 */
+	protected function get_list_of_polyfills(): array {
+		return self::LIST_OF_POLYFILLS;
+	}
+
+	/**
+	 * Remove PHP 8.0 specific code of Symfony polyfills
+	 *
+	 * @param Event $event Event context provided by Composer
+	 */
+	public static function remove( Event $event = null ) {
+		$cleanup = new SymfonyPolyfillCleanup( $event );
+		$cleanup->remove_bootstraps80();
+
+	}
+
+	/**
+	 * Remove bootstrap80.php files and the conditional statement in bootstrap.php
+	 */
+	protected function remove_bootstraps80() {
+		$this->output_text( 'Removing PHP 8.0 specific code for symfony polyfills' );
+
+		$list_of_polyfills = $this->get_list_of_polyfills();
+
+		foreach ( $list_of_polyfills as $polyfill ) {
+
+			// Search for bootstrap.php
+			$bootstraps = array_unique(
+				$this->find_library_file_pattern( 'bootstrap', $polyfill )
+			);
+
+			foreach ( $bootstraps as $bootstrap ) {
+				$this->remove_statement_with_pattern( $bootstrap, "return require __DIR__.'/bootstrap80.php", 'if (\PHP_VERSION_ID >= 80000)' );
+			}
+
+			// Search for bootstrap80.php
+			$bootstraps80 = array_unique(
+				$this->find_library_file_pattern( 'bootstrap80', $polyfill )
+			);
+
+			foreach ( $bootstraps80 as $bootstrap80 ) {
+				$this->remove_file( $bootstrap80 );
+			}
+		}
+	}
+
+
+	/**
+	 * Remove a statement from a file.
+	 *
+	 * @param string $file
+	 * @param string $inner_pattern The inner pattern for the  statement.
+	 * @param string $statement_pattern The statement pattern.
+	 */
+	protected function remove_statement_with_pattern( string $file, string $inner_pattern, string $statement_pattern ) {
+		if ( ! file_exists( $file ) ) {
+			$this->warning( sprintf( 'File does not exist: %s', $file ) );
+			return;
+		}
+
+		$contents = file_get_contents( $file ); // phpcs:ignore WordPress.WP.AlternativeFunctions
+
+		$pattern = '/' . preg_quote( $inner_pattern, '/' ) . '/';
+		if ( ! preg_match( $pattern, $contents, $matches, PREG_OFFSET_CAPTURE ) ) {
+			$this->warning( sprintf( 'Inner pattern %s not found in %s', $inner_pattern, $file ) );
+			return;
+		}
```

Comment:
I would have expected just searching for the if block would have been enough, but the double check to make sure it contains the right import statement works as well.

Also since inner pattern is a regular expression pattern I'd expect it to be something a little more flexible like:
`require .*/bootstrap80.php`
Either one is fine though, since they are consistent across packages.